aws/resource_aws_batch_job_definition.go

```diff
@@ -0,0 +1,209 @@
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{batch.JobDefinitionTypeContainer}, true),
+			},
+			"revision": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsBatchJobDefinitionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+	name := d.Get("name").(string)
+
+	input := &batch.RegisterJobDefinitionInput{
+		JobDefinitionName: aws.String(name),
+		Type:              aws.String(d.Get("type").(string)),
+	}
+
+	if v, ok := d.GetOk("container_properties"); ok {
+		props, err := expandBatchJobContainerProperties(v.(string))
+		if err != nil {
+			return fmt.Errorf("%s %q", err, name)
+		}
+		input.ContainerProperties = props
+	}
+
+	if v, ok := d.GetOk("parameters"); ok {
+		input.Parameters = expandJobDefinitionParameters(v.(map[string]interface{}))
+	}
+
+	if v, ok := d.GetOk("retry_strategy"); ok {
+		input.RetryStrategy = expandJobDefinitionRetryStrategy(v.(*schema.Set))
+	}
+
+	out, err := conn.RegisterJobDefinition(input)
+	if err != nil {
+		return fmt.Errorf("%s %q", err, name)
+	}
+	d.SetId(*out.JobDefinitionArn)
+	d.Set("arn", out.JobDefinitionArn)
+	return resourceAwsBatchJobDefinitionRead(d, meta)
+}
+
+func resourceAwsBatchJobDefinitionRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+	arn := d.Get("arn").(string)
+	job, err := getJobDefinition(conn, arn)
+	if err != nil {
+		return fmt.Errorf("%s %q", err, arn)
+	}
+	d.Set("arn", *job.JobDefinitionArn)
+	d.Set("container_properties", *job.ContainerProperties)
+	d.Set("parameters", job.Parameters)
+	d.Set("retry_strategy", flattenRetryStrategy(job.RetryStrategy))
+	d.Set("revision", *job.Revision)
+	d.Set("type", job.Type)
+	return nil
+}
+
+func resourceAwsBatchJobDefinitionDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+	arn := d.Get("arn").(string)
+	_, err := conn.DeregisterJobDefinition(&batch.DeregisterJobDefinitionInput{
+		JobDefinition: aws.String(arn),
+	})
+	if err != nil {
+		return fmt.Errorf("%s %q", err, arn)
+	}
+	d.SetId("")
+	return nil
+}
+
+func getJobDefinition(conn *batch.Batch, arn string) (*batch.JobDefinition, error) {
+	describeOpts := &batch.DescribeJobDefinitionsInput{
+		JobDefinitions: []*string{aws.String(arn)},
+	}
+	resp, err := conn.DescribeJobDefinitions(describeOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	numJobDefinitions := len(resp.JobDefinitions)
+	switch {
+	case numJobDefinitions == 0:
+		log.Printf("[DEBUG] Job Definition %q is already gone", arn)
+		return nil, nil
+	case numJobDefinitions == 1:
+		return resp.JobDefinitions[0], nil
```

Comment:
Shouldn't we also check the status of the definition here, since we treat `INACTIVE` definition as "deleted" I think it should be reflected here too, unless there's a strong reason not to do it.